src/sequence/mod.rs

```diff
@@ -719,22 +723,162 @@ mod tests {
+
+            let actor1 = gen_public_key();
+            let sequence_name = XorName::random();
+
+            let sdata_tag = 43_001u64;
+
+            // Instantiate the same Sequence on two replicas with two diff actors
+            let mut replica1 = Sequence::new_public(actor1, sequence_name, sdata_tag);
+            let mut replica2 = Sequence::new_public(actor1, sequence_name, sdata_tag);
+
+            // Set Actor1 as the owner
+            let perms = BTreeMap::default();
+            let owner_op = replica1.set_public_policy(actor1, perms)?;
+            replica2.apply_public_policy_op(owner_op)?;
+
+            let dataset_length = dataset.len() as u64;
+
+            // insert our data at replicas
+            for data in dataset {
+                let data = data.into_bytes();
+                // Append an item on replica1
+                let append_op1 = replica1.append(data.clone())?;
+                let append_op2 = replica2.append(data)?;
+
+                replica1.apply_data_op(append_op1)?;
+                replica2.apply_data_op(append_op2)?;
+            }
+
+            verify_data_convergence(vec![replica1, replica2], dataset_length);
+
+        }
+
+
+        #[test]
+        fn proptest_seq_converge_with_many_random_data_across_arbitrary_number_of_replicas( dataset in prop::collection::vec("\\PC*", 1..500), number_of_replicas in 1..50 ) {
+
+            let dataset_length = dataset.len() as u64;
+            let sequence_name = XorName::random();
+            let tag = 43_001u64;
+            let mut replicas = generate_replicas(number_of_replicas, sequence_name, tag)?;
+
+            // insert our data at replicas
+            for data in dataset {
+                let data = data.into_bytes();
+
+                for replica in &mut replicas {
+                    let append_op = replica.append(data.clone())?;
+
+                    replica.apply_data_op(append_op)?;
+
+                }
+            }
+
+            verify_data_convergence(replicas, dataset_length);
+
+        }
+
+
+        #[test]
+        fn proptest_converge_with_shuffled_data_set_across_arbitrary_number_of_replicas( dataset in prop::collection::vec("\\PC*", 1..100), number_of_replicas in 1..500 ) {
+
+            let dataset_length = dataset.len() as u64;
+
+            let sequence_name = XorName::random();
+            let tag = 43_001u64;
+            let mut replicas = generate_replicas(number_of_replicas, sequence_name, tag)?;
+
+
+            // insert our data at replicas
+            for data in dataset {
+                let data = data.into_bytes();
+
+                // first we generate standard data ops
+                for replica in &mut replicas {
+                    let append_op = replica.append(data.clone())?;
+                    let mut ops = vec![];
+                    ops.push(append_op);
+                    ops.shuffle(&mut OsRng);
```

Comment:
I think the intention to build up ops in the ops vec, right? In that case the instantiation just needs to be moved out of the `for replica` loop.

Reply:
At the moment, I was just intending to test shuffling ops for _that replica_ only. 

If we move it out, we'd get an `ops` of all ops across all replicas, which would fail when trying to apply ops to another replica.

`Test failed: Data operation depends on a different replica's state than the current;`

Reply:
This may be me misunderstanding how we should be applying this stuff though (looking at gabs other comments), so perhaps what you're suggesting _should_ work...

Reply:
Grand, have this set up now, w/ one op set generated for one replica, and then that is shuffled + applied for each replica thereafter :+1: